lib/IRGen/GenProto.cpp

```diff
@@ -981,16 +981,17 @@ static bool isDependentConformance(IRGenModule &IGM,
       return true;
   }
 
+  auto DC = conformance->getDeclContext();
   // If the conforming type isn't dependent, the below check is never true.
-  if (!conformance->getDeclContext()->isGenericContext())
+  if (!DC->isGenericContext())
     return false;
 
   // Check whether any of the associated types are dependent.
   if (conformance->forEachTypeWitness(nullptr,
         [&](AssociatedTypeDecl *requirement, Type type,
             TypeDecl *explicitDecl) -> bool {
           // RESILIENCE: this could be an opaque conformance
-          return type->hasArchetype();
+          return DC->mapTypeIntoContext(type)->hasArchetype();
```

Comment:
This can just be `type->hasTypeParameter()`?

Reply:
I think so!